src/model.ts

```diff
@@ -1,13 +1,37 @@
+require('isomorphic-fetch');
+import * as _ from 'lodash';
 import { Request } from 'express';
-import { Readable } from 'stream';
+import { PagingStream } from './paging-stream';
+import { searchContent } from '@esri/hub-search';
+import { IContentSearchRequest } from './types';
 
 export class HubApiModel {
-  async getStream (request: Request) {
-    console.log(request);
 
-    const ret = new Readable({ read () {} });
-    ret.push({ foo: 'bar' });
-    ret.push(null);
-    return ret;
+  getStream (request: Request) {
+    const searchRequest: IContentSearchRequest = request.res.locals.searchRequest;
+
+    // TODO support multiple environments
+    _.set(searchRequest, 'options.portal', 'https://www.arcgis.com');
+
+    const searchApiStream = new PagingStream({
+      firstPageParams: searchRequest,
+
+      loadPage: (params: IContentSearchRequest | any) => {
+        if (params.next) {
+          return params.next();
+        }
+
+        return searchContent(params); // first page request
```

Comment:
@dbouwman I've refactored to use the `searchContent` function. Thanks for this suggestion.